src/test/java/uk/gov/ch/oauth/Oauth2Test.java

```diff
@@ -61,9 +72,21 @@
     @Mock
     private Session session;
     @Mock
+    private Cookie cookie;
+    @Mock
     private SessionFactory sessionFactory;
     @Mock
     private HttpServletRequest request;
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private OAuth2StateHandler oAuth2StateHandler;
+    @Mock
+    private Payload payload;
+    @Mock
+    private JSONObject jsonObject;
+    @Mock
+    private HttpServletResponse httpServletResponse;
+    @Spy
+    private Map<String, Object> data;
```

Comment:
Data is a very common variable name in this, could we rename the field level to make it more clear what it does.